### PR TITLE
Update graphql testing

### PR DESCRIPTION
**Pull request by dahukish**
## Description

Fixes (issue https://github.com/Shopify/web/issues/59428)

Bumps graphql-testing to apollo client v3. 

Bonus bumps apollo/client to v3 in the main package.json

## Type of change

- [ ]  Patch: Bug (non-breaking change which fixes an issue)
- [ ]  Minor: New feature (non-breaking change which adds functionality)
- [x]  Major: Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [ ] I have added a changelog entry, prefixed by the type of change noted above (Documentation fix and Test update does not need a changelog as we do not publish new version)

### CODE REVIEW

File: packages/graphql-testing/src/client.ts [hunk 1]
```diff
@@ -1,67 +0,0 @@
-import {
```
Comment (emiryy) [inline packages/graphql-testing/src/client.ts:1]: Non blocking, just curious why this file is no longer needed

File: packages/react-graphql/src/async/tests/component.test.tsx [hunk 2]
```diff
@@ -143,7 +143,7 @@ describe('createAsyncQueryComponent()', () => {
 
       const client = createMockApolloClient();
       const watchQuerySpy = jest.spyOn(client, 'watchQuery');
-      watchQuerySpy.mockImplementation(() => ({subscribe() {}} as any));
+      watchQuerySpy.mockImplementation(() => ({subscribe() {} } as any));
```
Comment (emiryy) [inline packages/react-graphql/src/async/tests/component.test.tsx:146]: This seems to be extra spacing which will cause the linter to fail. See below on like 205 for same thing.

**Review by emiryy — APPROVED** [review]
Looks good to me!

File: packages/graphql-testing/src/client.ts (hunk 1 above)
Comment (dahukish) [reply packages/graphql-testing/src/client.ts:1]: This was added two years ago to stop certain events happening out of `react` `act`. We will need to reevaluate and figure out if this is even needed anymore. I am leaning towards not.

**Pull request merged**